hibernate-core/src/main/java/org/hibernate/id/uuid/UuidV6ValueGenerator.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.id.uuid;
+
+import java.security.SecureRandom;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.id.UUIDGenerationStrategy;
+
+/**
+ * Implements a <a href="https://datatracker.ietf.org/doc/html/rfc9562#name-uuid-version-6">UUID Version 6</a> generation strategy as defined by the {@link UUID#randomUUID()} method.
+ *
+ * @author Cedomir Igaly
+ */
+public class UuidV6ValueGenerator implements UUIDGenerationStrategy, UuidValueGenerator {
+
+	private static final Instant EPOCH_1582;
+
+	static {
+		EPOCH_1582 = LocalDate.of( 1582, 10, 15 )
+				.atStartOfDay( ZoneId.of( "UTC" ) )
+				.toInstant();
+	}
+
+	private static class Holder {
+
+		static final SecureRandom numberGenerator = new SecureRandom();
+	}
+
+	public static final UuidV6ValueGenerator INSTANCE = new UuidV6ValueGenerator();
+
+	private final Lock lock = new ReentrantLock( true );
+
+	private long currentTimestamp;
+
+	private final AtomicLong clockSequence = new AtomicLong( 0 );
+
+	public UuidV6ValueGenerator() {
+		this( getCurrentTimestamp(), 0 );
+	}
+
+	public UuidV6ValueGenerator(final long currentTimestamp, final long clockSequence) {
+		this.currentTimestamp = currentTimestamp;
+		this.clockSequence.set( clockSequence );
+	}
+
+	/**
+	 * A variant 6
+	 */
+	@Override
+	public int getGeneratedVersion() {
+		// UUIDv6 is a field-compatible version of UUIDv1, reordered for improved DB locality
+		return 6;
+	}
+
+	/**
+	 * Delegates to {@link UUID#randomUUID()}
```

Comment:
Not sure why you are referencing `UUID#randomUUID` here?

Reply:
Same mistake as on other three places